src/main.rs

```diff
@@ -355,44 +349,95 @@ async fn attest(
 
     // Generate the instance id.
     let uuid = uuid::Uuid::new_v4();
+    let serial_number = UIntRef::new(uuid.as_bytes()).or(Err(StatusCode::INTERNAL_SERVER_ERROR))?;
 
-    // Create the new certificate.
-    let tbs = TbsCertificate {
+    let signature = pki
+        .signs_with()
+        .or(Err(StatusCode::INTERNAL_SERVER_ERROR))?;
+
+    // Create and sign the new certificate.
+    TbsCertificate {
         version: x509::Version::V3,
-        serial_number: UIntRef::new(uuid.as_bytes()).or(Err(ISE))?,
-        signature: isskey.signs_with().or(Err(ISE))?,
+        serial_number,
+        signature,
         issuer: issuer.tbs_certificate.subject.clone(),
-        validity,
+        validity: *validity,
         subject: RdnSequence(Vec::new()),
-        subject_public_key_info: cri.public_key,
+        subject_public_key_info: info.public_key,
         issuer_unique_id: issuer.tbs_certificate.subject_unique_id,
         subject_unique_id: None,
         extensions: Some(extensions),
-    };
+    }
+    .sign(pki)
+    .or(Err(StatusCode::INTERNAL_SERVER_ERROR))
+}
+
+/// Receives:
+/// ASN.1 SEQUENCE OF CertRequest.
+/// Returns:
+/// ASN.1 SEQUENCE OF Output.
+async fn attest(
+    body: Bytes,
+    Extension(state): Extension<Arc<State>>,
+) -> Result<Vec<u8>, impl IntoResponse> {
+    // Decode the signing certificate and key.
+    let issuer = Certificate::from_der(&state.crt).or(Err(StatusCode::INTERNAL_SERVER_ERROR))?;
+    let isskey = PrivateKeyInfo::from_der(&state.key).or(Err(StatusCode::INTERNAL_SERVER_ERROR))?;
 
-    // Sign the certificate.
-    let crt = tbs.sign(&isskey).or(Err(ISE))?;
-    let crt = Certificate::from_der(&crt).or(Err(ISE))?;
+    let now = SystemTime::now();
+    let end = now + Duration::from_secs(60 * 60 * 24 * 28);
```

Comment:
I added a constant here in https://github.com/rjzak/steward/pull/1, can we bring it back?
The reason is to avoid magic numbers